tests/devices/default_qubit/test_default_qubit.py

```diff
@@ -1486,6 +1494,34 @@ def test_finite_shots_postselection_defer_measurements(self, max_workers):
         first = next(iterator)
         assert all(np.array_equal(first, rest) for rest in iterator)
 
+    def test_integrate_prng_key_jitting(self):
+        """Test that a PRNGKey can be used with a jitted qnode."""
+
+        import jax
+
+        @jax.jit
+        def workflow(key, param):
+
+            dev = qml.device("default.qubit", seed=key, shots=100)
+
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(x, 0)
+                return qml.sample(wires=0)
+
+            return circuit(param)
+
+        key1 = jax.random.PRNGKey(123456)
+        key2 = jax.random.PRNGKey(8877655)
+        x = jax.numpy.array(0.5)
+        # no leaked tracer errors
+        res1 = workflow(key1, x)
+        res1_again = workflow(key1, x)
+        res2 = workflow(key2, x)
+
+        assert qml.math.allclose(res1, res1_again)
+        assert not qml.math.allclose(res1, res2)
```

Comment:
This is fine for now, but just a note that JAX does not guarantee PRNG key outputs to be consistent between releases. Maybe worth adding a really low tolerance here to drive home the fact that different results are expected for different seeds.

Reply:
The chance that different keys gives the exact same raw samples would be `0.5**100 = 7e-31`.  I think we can safely say that that the samples will always be independent as long as the keys are different.